actor/v7action/service_instance_sharing.go

```diff
@@ -0,0 +1,50 @@
+package v7action
+
+import (
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv3"
+	"code.cloudfoundry.org/cli/resources"
+	"code.cloudfoundry.org/cli/types"
+	"code.cloudfoundry.org/cli/util/railway"
+)
+
+type ServiceInstanceSharingParams struct {
+	SpaceName string
+	OrgName   types.OptionalString
+}
+
+func (actor Actor) ShareServiceInstanceToSpaceAndOrg(
+	serviceInstanceName, targetedSpaceGUID, targetedOrgGUID string,
+	sharedToDetails ServiceInstanceSharingParams,
+) (Warnings, error) {
+	var shareToOrgGUID = targetedOrgGUID
+
+	return handleServiceInstanceErrors(railway.Sequentially(
+		func() (warnings ccv3.Warnings, err error) {
+			_, _, warnings, err = actor.CloudControllerClient.GetServiceInstanceByNameAndSpace(serviceInstanceName, targetedSpaceGUID)
+			return
+		},
+		func() (warnings ccv3.Warnings, err error) {
+			if sharedToDetails.OrgName.IsSet {
+				var (
+					orgWarnings  Warnings
+					organization resources.Organization
+				)
+
+				organization, orgWarnings, err = actor.GetOrganizationByName(sharedToDetails.OrgName.Value)
+				warnings = ccv3.Warnings(orgWarnings)
+				if err == nil {
```

Comment:
Not sure I understand the part that is not needed. 
If it is the err `GetOrganizationByName` returns an error when the org list is empty

Reply:
The `if err == nil` is the bit I pointed out as unnecessary. Even when `GetOrganizationByName` errors, the organization will be initialised (with empty, yes, but it will nonetheless).

If we expand the code we have here with the railways code, we end up with something like:

```
organization, orgWarnings, err = actor.GetOrganizationByName(sharedToDetails.OrgName.Value)
if err == nil {
  // do something
}
if err != nil {
  return err, warning
}
```